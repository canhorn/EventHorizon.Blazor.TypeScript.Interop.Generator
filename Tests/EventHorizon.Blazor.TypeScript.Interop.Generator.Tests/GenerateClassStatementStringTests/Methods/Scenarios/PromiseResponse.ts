declare module Static.Class {
  export class ExampleClass<T> {
    promiseVoid(): Promise<void>;
    promiseVoidVoid(): Promise<void[]>;
  }
}
