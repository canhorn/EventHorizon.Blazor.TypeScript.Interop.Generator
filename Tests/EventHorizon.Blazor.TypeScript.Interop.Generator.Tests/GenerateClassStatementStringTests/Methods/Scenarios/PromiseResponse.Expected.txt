/// Generated - Do Not Edit
namespace Static.Class
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass<CachedEntity>>))]
    public class ExampleClass<T> : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public async ValueTask promiseVoid()
        {
            await EventHorizonBlazorInterop.Task<Void_>(
                new object[]
                {
                    new string[] { this.___guid, "promiseVoid" }
                }
            );
        }

        public async ValueTask promiseVoidVoid()
        {
            EventHorizonBlazorInterop.TaskArray<Void_>(
                new object[]
                {
                    new string[] { this.___guid, "promiseVoidVoid" }
                }
            );
        }
        #endregion
    }
}